### PR TITLE
Statusreply update

### PR DESCRIPTION
**Pull request by pysailor**
Makes it possible to post comments on status updates.
See also cosent/plonesocial.activitystream/pull/33

### CODE REVIEW

File: src/plonesocial/microblog/browser/tiles/newpostbox.py [hunk 1]
```diff
@@ -40,6 +40,8 @@ def post_context(self):
         ''' The context of this microblog post
         (the portal, a workspace, and so on...)
         '''
+        if self.thread_id:
+            return self.post_container.get(self.thread_id)
         return get_microblog_context(self.context)
```
Comment (gyst) [inline src/plonesocial/microblog/browser/tiles/newpostbox.py]: This looks invalid. If you have a thread_id it returns an object of type IStatusUpdate (a tweet). If you don't have a thread id it returns an IMicroblogContext (workspace) or ISiteRoot.

One cannot assume that IStatusUpdate us usable as a "normal" context without doing the extensive analysis and refactoring we have planned for the coming weeks.

Comment (pysailor) [inline src/plonesocial/microblog/browser/tiles/newpostbox.py]: @gyst Hm, the idea behind this is: If we have a thread id, then we're posting in the context of a StatusUpdate. That means, the context for the new post is said StatusUpdate. So in other words, we're posting a reply on an existing StatusUpdate. We're not using it as "normal" context, but just as the context in which the message currently being posted lands.

If we don't have a thread id, then it's a new StatusUpdate on its own.

For the case of the activity stream that we show in dashboard and workspace, this seems to do exactly what we want. But I agree, there might lots more under the hood that I don't understand yet.
=> Discuss during the refactoring session.

File: src/plonesocial/microblog/tests/test_tiles.py [hunk 2]
```diff
@@ -71,3 +73,56 @@ def test_newpostbox_tile_submission_on_portal(self):
...
+        # the tile works in the context os the previous status update
+        self.assertEqual(tile.post_context.id, thread_id)
+        # that has no repies
+        replies = tuple(tile.post_context.replies())
+        self.assertEqual(len(replies), 0)
+        # until the tile is processed
+        tile.update()
+        # after the async machinery has finished
+        # we have one reply
+        sleep(2)
```
Comment (ale-rt) [inline src/plonesocial/microblog/tests/test_tiles.py]: Instead of sleeping we should call the flush method for this queue container

**Comment by gyst** [comment]
Cancelled

**Pull request closed without merging**